crypto/async/arch/async_win.c

```diff
@@ -20,6 +20,21 @@ int ASYNC_is_capable(void)
     return 1;
 }
 
+int ASYNC_set_mem_functions(ASYNC_stack_alloc_fn alloc_fn,
+                            ASYNC_stack_free_fn free_fn)
+{
+    return 0;
+}
+
+void ASYNC_get_mem_functions(ASYNC_stack_alloc_fn *alloc_fn,
+                             ASYNC_stack_free_fn *free_fn)
+{
+    if (alloc_fn)
+        *alloc_fn = NULL;
+    if (free_fn)
+        *free_fn = NULL;
+}
```

Comment:
I'm wondering why this functionality isn't available to windows users. It seems there is nothing posix specific in the posix implementation.

Reply:
The API documentation for `CovertThreadToFibreEx` is available here: https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-convertthreadtofiberex and an overview of Windows Fibers is available here: https://docs.microsoft.com/en-us/windows/win32/procthread/fibers.

It appears Windows handles allocating the stack for fibers internally and the caller has little/no control on how that occurs.

On the POSIX side `makecontext` requires the caller to explicitly allocate stack space.

Because the user appears to have no control over stack allocation for Fibers on Windows `ASYNC_set_mem_functions` and `ASYNC_get_mem_functions` are non-functional stubs on that platform.

Reply:
Ah - right of course. That makes sense.